src/rules/padding.ts

```diff
@@ -0,0 +1,336 @@
+/**
+ * @fileoverview Rule to require or disallow newlines between jest functions
+ *   Based on eslint/padding-line-between-statements by Toru Nagashima
+ *   See: https://github.com/eslint/eslint/blob/master/lib/rules/padding-line-between-statements.js
+ *
+ *   STATEMENT_LIST_PARENTS, isTokenOnSameLine and isSemiColonToken borrowed from
+ *   eslint ast-utils by Gyandeep Singh
+ *   See: https://github.com/eslint/eslint/blob/master/lib/rules/utils/ast-utils.js
+ */
+import { AST, Rule, SourceCode } from 'eslint';
+import { Node } from 'estree';
+
+const STATEMENT_LIST_PARENTS = new Set([
+  'Program',
+  'BlockStatement',
+  'SwitchCase',
+  'SwitchStatement',
+]);
+
+const isTokenOnSameLine = (
+  left: Node | AST.Token,
+  right: Node | AST.Token,
+): boolean => left.loc.end.line === right.loc.start.line;
+
+const isSemicolonToken = (token: AST.Token): boolean =>
+  token.value === ';' && token.type === 'Punctuator';
+
+// NOTE: Should probably be configurable
+const isTestFile = (filename: string): boolean =>
+  filename.includes('test') || filename.includes('spec');
+
+/**
+ * Creates tester which check if an expression node has a certain name
+ *
+ * @returns {Object} the created tester.
```

Comment:
Still working on a few types here...

Reply:
If you feel its onerous, don't worry about it